### DISCUSSION TITLE
Keyball 61 Right and leftside not working together

### DISCUSSION THREAD

**Discussion by eeJoe** [Yowkees/keyball#480] [Q&A]
Hoping somebody can help with a few issues I am having at the moment please.

Issue #1: LED's on the right side not working.
I believe I have isolated the issue to some degree, when I test with multimeter on LED no.1, the bottom plates don't give me a beep, this is consistent across # 2 and #3 but #34 and #33 seem to work?
I have replaced the LED with a new one but the issue seem to persist, is there anything I am missing here?

Issue#2: OLED screen on the right side not working.
This issue was there eversince I put together the keyboard with initial parts purchased. I thought this may be due to faulty OLED screen / ProMicroboard so replaced both with a new part. However, the issue seem to persist, which makes me think this could be due to something else? Kindly let me know if I am missing anything else, thanks.

Issue #3: TRS / TRRS link does not work
Both sides work fine when I test individually (without TRRS cable connected).
However, when I link the two keyboards with TRRS cable and plug in the usb into either proMicroboard, only one of the sides work (either left or right, but not both).
I have tried with 3 different cables, which all yields same results. To be honest, I am happy to live with the above two issues if the keyboard would at least work together...

Would really appreciate any help on this please, thank you!

Best,

Joe
![View recent photos](https://github.com/Yowkees/keyball/assets/64901119/b23428f4-39f3-48eb-a079-e4c292c71744)
![Screenshot 2024-02-05 at 21 05 42](https://github.com/Yowkees/keyball/assets/64901119/8fc432b5-14d9-4307-962d-11452e1c1549)

**Comment by Yowkees** [comment]
Hi Joe,

Thank you for choosing Keyball61.

For catch up #3 problem,
I think there is some problem of soldering on TRS connecter or ProMicro.
Please heat PCB silver pad enough and add fresh solder again.

If this catch up doesn't work, please show me the picture of soldering face.

Thank you very much.

**Reply by eeJoe** [reply]
I'll give this a shot, thanks Yowkees!

**Reply by eeJoe** [reply]
Hi Yowkees,
I have gone through soldering of TRS connector and ProMicro board, but the same issue still persists & honestly not sure whereelse to look.

![IMG_6270](https://github.com/Yowkees/keyball/assets/64901119/8e92616d-04b3-4711-87af-7720da255db8)
![IMG_6269](https://github.com/Yowkees/keyball/assets/64901119/f620dad1-d6a5-4b0e-90f8-0f55988e4733)
![IMG_6267](https://github.com/Yowkees/keyball/assets/64901119/590eb2f9-c97c-4caf-bef2-f46b58005d18)
![IMG_6268](https://github.com/Yowkees/keyball/assets/64901119/758508dc-150b-4b14-998d-f26d290579e4)
Please find the attached photos. Would greatly appreciate if you can please have a look and let me know what might have gone wrong.

**Reply by eeJoe** [reply]
@Yowkees - appreciate if you can have a look at the above please, thanks!

**Reply by eeJoe** [reply]
Hi @Yowkees - would you be able to let me know if anything looks odd from the above that may be causing issue #3 please?

**Reply by eeJoe** [reply]
@koron - I have raised this ticket a while ago but still suffering from the issue you have replied to in another ticket 
In response to your comment in the other thread (https://github.com/Yowkees/keyball/discussions/509#discussioncomment-8963010), here are my results:

1.  Tested with these firmwares previously, same issue persists regardless of firmware version 
![image](https://github.com/Yowkees/keyball/assets/64901119/48a8e403-48ac-4c9a-9e28-7f7413657f6a)
2. Updated to the latest version - issue still persists
![image](https://github.com/Yowkees/keyball/assets/64901119/c9b0ec92-16ee-49d4-997f-0378c6d70cef)

3. Yes, it works fine independently.
4. Yes, it works fine independently (without ball).
5. Yes, works fine with TRRS cable inserted in only one side (ball).
6. Yes, works fine with TRRS cable inserted in only one side (without ball).
7. No ball side displays "Keyball 61", unsure what the ball side displays... LED on my ballside does not work (I've replaced it & Promicro board, however it does not work). Only ball side keyboard & ball works.
8. No ball side displays the following message: Only this keyboard works and not the one with ball.
Key: R0 C0
Ball: 0 0 0 0
CPI 500 S0 D4
Layer:_______________

**Reply by koron** [reply]
@eeJoe

Could you please measure the voltage between GND and VCC?

If it doesn't seem to be above 4V, try shorting the open jumper on the ProMicro with solder. 
The jumper location is next to Type-C receptacle.
So it should be more than 4.5V.

There are some Pro Micro compatible devices only supply 3.3V even though they are driven at 16MHz, and may cause a power shortage.
I have determined that the reported symptoms are very similar to those caused by a lack of power.

**Reply by eeJoe** [reply]
Thank you so much for looking into this @koron - I can see that the reading is 4.61 - 4.62V between GND and VCC on both sides of keyboard without TRS link.

It ranges 4.0 - 4.3 V when I link two keyboards together... so doesn't seem like it's the issue with voltage going below 4V in any one keyboard.

I've checked voltage on all three GND pads vs VCC on both keyboards (with TRS link & without).

Would there be anything else I should be checking?

Appreciate your help on this.

Best,

Joe

**Comment by Yowkees** [comment]
Hi Joe,

I apologize my delay.

It could be mischief of light, but please make sure green circled parts are not shorted.
If it's shorted, please reheat here and separate.

![61Rfix](https://github.com/Yowkees/keyball/assets/84209772/6bcc44b2-6b2c-4e00-92f6-9ecd9a29d6ef)

If LEDs are not still work, go next step.
Please try grind Red line on above picture by the back of box cutter (2nd picture).
When this part's Right and Left pads are separated, LED data line is off.
This experiment shows whether LED is broken by short mode.

![cutter](https://github.com/Yowkees/keyball/assets/84209772/774747b4-f43c-4b1d-b3aa-85015644cc37)

Thank you very much!

**Reply by eeJoe** [reply]
Thank you @Yowkees,

Can you please have a look if I’ve done it right? Am i meant to see any difference after grinding down the Led data line?

I can confirm there are no shorts between the green circles as per the attached. 

Appreciate your help.

Joe

**Comment by eeJoe** [comment]
![IMG_6382](https://github.com/Yowkees/keyball/assets/64901119/0a3d298c-b022-42d3-b8fc-43dd25a75833)

**Reply by eeJoe** [reply]
Hi @Yowkees - appreciate if you can get back to me on the above please, thanks.

**Reply by eeJoe** [reply]
Hi @Yowkees & @koron! 

Hope you are well, just wanted to touch base on the above please.

Best,

Joe

**Reply by eeJoe** [reply]
Hi @Yowkees & @koron,

Hope you are well. Just wanted to touch base on the above, would you have any other suggestions? i really just want to be able to use the keyboards at this stage.

Best,

Joe

**Reply by eeJoe** [reply]
Hi @Yowkees & @koron, just wanted to check on the above queries thanks.

**Reply by eeJoe** [reply]
@Yowkees & @koron - please don't leave me hanging bros

**Reply by eeJoe** [reply]
@Yowkees & @koron - hi gents, any update on this would be appreciated thanks.

**Reply by eeJoe** [reply]
@Yowkees & @koron - bros..?
